test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,325 @@
+            self.assertTrue(rv)
+
+        # Start the previous inv mom.
+        self.inv_mom.start()
+
+        # Check previous inventory mom is up
+        rv = self.server.expect(
+            VNODE, {'state': 'free'}, id=self.inv_mom.shortname,
+            max_attempts=3, interval=2)
+        self.assertTrue(rv)
+
+    def test_two_moms_different_vnode_pool(self):
+        """
+        Differing vnode_pool for two moms shall result in both moms reporting
+        inventory.
+        """
+        found_in_momA = 0
+        found_in_momB = 0
+        self.momA = self.moms.values()[0]
+        try:
+            self.momB = self.moms.values()[1]
+        except IndexError:
+            self.assertTrue(
+                False, msg="Provide at least 2 moms while invoking test")
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr_A = {'vnode_pool': '1'}
+        attr_B = {'vnode_pool': '2'}
+
+        start_time = int(time.time())
+
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA, attrib=attr_A)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB, attrib=attr_B)
+
+        rc = self.momA.log_match(
+            "Hello (no inventory required) from server", max_attempts=5,
+            starttime=start_time)
+        if rc is None:
+            found_in_momA = 0
+        else:
+            found_in_momA = 1
+
+        rc = self.momB.log_match(
+            "Hello (no inventory required) from server", max_attempts=5,
+            starttime=start_time)
+        if rc is None:
+            found_in_momB = 0
+        else:
+            found_in_momB = 1
+
+        if (found_in_momA + found_in_momB != 0):
+            self.assertTrue(False, msg="Both moms must report inventory")
+
+    def test_invalid_usage(self):
+        """
+        Setting vnode_pool for an existing mom that does not have a vnode_pool
+        attribute shall not be allowable.
+        Setting vnode_pool for an existing mom having a vnode_pool attribute
+        shall not be allowable.
+        Unsetting vnode_pool for an existing mom having a vnode_pool attribute
+        shall not be allowable.
+        """
+        self.momA = self.moms.values()[0]
+        self.hostA = self.momA.shortname
+        self.logger.info("hostA is %s." % self.hostA)
+
+        start_time = int(time.time())
+
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+
+        attr_2 = {'vnode_pool': '2'}
+        try:
+            self.server.manager(
+                MGR_CMD_SET, NODE, id=self.hostA, attrib=attr_2, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15004) returned from server")
+
+        rv = self.server.log_match(
+            "Unsupported actions for vnode_pool", max_attempts=5,
+            starttime=start_time)
+        self.assertTrue(rv, msg="Found correct server log message")
```

Comment:
the msg will only print if the assert fails.  The message sounds like it passed.